packages/eth-json-rpc-provider/src/safe-event-emitter-provider.test.ts

```diff
@@ -122,3 +168,83 @@ describe('SafeEventEmitterProvider', () => {
     });
   });
 });
+
+describe('convertEip1193RequestToJsonRpcRequest', () => {
+  it('converts an EIP-1193 request to a JSON-RPC request', () => {
```

Comment:
It seems like this test leaves `id` and `jsonrpc` out of the given request, but the tests below also do this, so it is a bit redundant.

I've also noticed that in the tests below ("it generates a unique id if id is not provided", etc.) we just check one property at a time. This makes sense, but it does mean that as a reader, if I want to ensure that this function is fully covered, I have to combine two tests together mentally, and it might be easier if each test fully tests the result of the function.

Perhaps we don't need this test, and instead, what we if test the complete value of `jsonRpcRequest` in each test below?